upload-api/did-resolver.js

```diff
@@ -0,0 +1,58 @@
+import { Schema, DIDResolutionError } from '@ucanto/validator'
+import { ok, error } from '@ucanto/core'
+import { parseDidPlc } from '@storacha/did-plc'
+import * as ed25519 from '@ucanto/principal/ed25519'
+
+/**
+ * @typedef {Record<`did:web:${string}`, `did:key:${string}`>} PrincipalMapping
+ */
+
+/**
+ * @typedef {import('@ucanto/interface').Signature} Signature
+ */
+
+/**
+ * Creates a DID resolver that can resolve web and PLC DIDs to their corresponding DID keys.
+ *
+ * @param {PrincipalMapping} principalMapping - Mapping of known WebDIDs to their corresponding DIDKeys
+ * @param {import('@storacha/did-plc').PlcClient} plcClient - PLC client to use for resolving PLC DIDs
+ * @returns {(did: `did:${string}:${string}`) => Promise<import('@ucanto/interface').Result<`did:key:${string}`[], DIDResolutionError>>}
+ */
+export function createDidResolver(principalMapping, plcClient) {
```

Comment:
Can these please be optional options? Would like to add a did:web resolver now we host DID documents for our services. None of the params should be required...